### PR TITLE
commonjs modules, for nodejs compatibility

### PR DESCRIPTION
**Pull request by massaroni**
## Description
Refactored all the es6 modules to be commonjs modules.

## Motivation and Context
This is a solution to [Issue #412](https://github.com/BrainJS/brain.js/issues/412), so that you can run example scripts in nodejs without running a build.  This should save time for contributors, and make it easier for nodejs users to work with brain.js as a dependency.

## Testing
I checked the consistency of the refactoring 3 different ways: 1) ran a dist build  2) ran example scripts in node, and 3) ran the existing unit tests.  Note that some tests are failing, but this is consistent with the develop branch, so at least there are no regressions in this PR.  And the example scripts are busted, but that's also consistent with the develop branch.

### CODE REVIEW

**Review by mubaidr — APPROVED** [review]

File: __tests__/layer/fully-connected.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-  predict3D,
-  compareBiases,
-  compareFilterDeltas,
-  compareFilterDeltas3D,
-  compareInputDeltas,
-  compareInputDeltas3D,
-} from '../../src/layer/fully-connected';
-import { onePlusPlus2D, zero2D } from '../test-utils';
+const gpuMock = require('gpu-mock.js');
+const fc = require('../../src/layer/fully-connected');
```
Comment (mubaidr) [inline __tests__/layer/fully-connected.js]: You can use Object Destructuring here to simplify requires. like this: 
```js
const { predict, predict3D, compareBiases, compareFilterDeltas, compareFilterDeltas3D, compareInputDeltas, compareInputDeltas3D } = require('../../src/layer/fully-connected');
```

File: __tests__/layer/leaky-relu.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
-import gpuMock from 'gpu-mock.js';
-import { predict, compare } from '../../src/layer/leaky-relu';
+const gpuMock = require('gpu-mock.js');
+const llr = require('../../src/layer/leaky-relu');
```
Comment (mubaidr) [inline __tests__/layer/leaky-relu.js]: Use Object Destructuring here too

**Review by mubaidr — COMMENTED** [review]
Thank you! Looks good. Just need few small improvements.

File: __tests__/layer/fully-connected.js (hunk 1 above)
Comment (massaroni) [reply __tests__/layer/fully-connected.js]: Got it, that would make it tighter.  Should I put that in a new commit and submit a new PR for that?  Or is there a way to append it to this PR?

Comment (mubaidr) [reply __tests__/layer/fully-connected.js]: Just push more commits to this branch, these will be automatically added to this pull request. 👍

**Comment by robertleeplummerjr** [comment]
You guys rock.

**Review by mubaidr — APPROVED** [review]
Looks good enough! 💯

**Comment by mubaidr** [comment]
Thanks @massaroni 

@robertleeplummerjr lots of tests are broken due to `gpujs` need to look into this issue.

**Comment by robertleeplummerjr** [comment]
@mubaidr would you lower your voice...  I'm working here!  Jk.  Ty for the prodding. I'm pushing some changes today for gpu.js, dev node and documentation, will probably have some time starting next weekey to really focus on it.

**Comment by mubaidr** [comment]
I would like to help you with this, let me know when you finish planned changes. I will then look into the tests.

**Comment by robertleeplummerjr** [comment]
ok, perfect!

**Comment by GabrielDelepine** [comment]
Let's undo this PR! ESM is now the standard

**Pull request merged**